### ISSUE TITLE
TinyMCE is failing WAVE accessibility plugin validation

### ISSUE THREAD

**Issue by Hauk** [tinymce/tinymce#4465]
**Do you want to request a *feature* or report a *bug*?**

Bug.

**What is the current behavior?**

TinyMCE is failing accessibility checks using the Wave accessibility plugin.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tinymce.com](http://fiddle.tinymce.com/) or similar.**

1. Install the WAVE Accessibility plugin: https://addons.mozilla.org/en-US/firefox/addon/wave-accessibility-tool/

2. Visit: https://www.tinymce.com/

3. Click the WAVE addon.

4. Observe that the editor has numerous buttons that are empty or do not have value text. This is breaking accessibility scores for a number of sites that use the TinyMCE editor.

![image](https://user-images.githubusercontent.com/243723/41976749-b4e9de86-7a15-11e8-9ea1-cfbc91ffee91.png)

**What is the expected behavior?**

TinyMCE should add the correct descriptive text so that a screen reader can indicate the function of a button to users.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**

Affected version is: TinyMce 4.3.17. I also tested with TinyMCE 4.2.4 and the same symptoms are occurring.

Can this be actioned to be fixed? Under the Americans with Disabilities Act, any company/institution operating a website which uses TinyMCE is potentially leaving them open to legal action. More information here:

https://www.newmediacampaigns.com/blog/website-accessibility-and-the-law-ada-wcag20

Thanks for the help with this!
Eoghan

**Comment by Afraithe** [comment]
I will have to look into this, but in general, straight testing tools run on TinyMCE doesn't fare well, there are a number of things happening in the UI that might seem strange to a testing tool such as Wave, but not when run with actual screen readers. We can't really trust these tools, so we do a great deal of manual testing as well as testing by actual ppl with disabilities to make sure it doesn't just conform to standards, but works in the real world.

**Comment by Hauk** [comment]
@Afraithe Thank you Afraithe!

**Comment by SteelWagstaff** [comment]
Thanks @Hauk & @Afraithe. We shared similar feedback in a WordPress ticket recently, and heard [a similar reply](https://core.trac.wordpress.org/ticket/47028#comment:2) from their a11y folks.

**Comment by thedrjim** [comment]
This issue has been closed due to inactivity. Tiny will soon be working with members of the community to close other old issues on our GitHub repositories.
Please note the official support window for TinyMCE 4 is ending on December 31, 2020. If you haven't already upgraded to TinyMCE 5 it is highly encouraged that you do so. Upgrading will give you access to the latest releases and updates made to our WYSIWYG editor focused on bringing the latest UX and developer features into the editing experience. To assist with upgrading the editor we have put together a migration guide you can use here - https://www.tiny.cloud/blog/how-to-migrate-from-tinymce-4-to-tinymce-5/
If you have any questions or concerns, you can contact me at community-qa@tiny.cloud. You can also join the discussion on our mailing list at the following link: community-qa+subscribe@tiny.cloud.
